pkg/cable/vxlan/vxlan.go

```diff
@@ -0,0 +1,537 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package vxlan
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/admiral/pkg/log"
+	v1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/cable"
+	"github.com/submariner-io/submariner/pkg/natdiscovery"
+	netlinkAPI "github.com/submariner-io/submariner/pkg/netlink"
+	"github.com/submariner-io/submariner/pkg/types"
+	"github.com/submariner-io/submariner/pkg/util"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+	"k8s.io/klog"
+)
+
+const (
+	VxlanIface             = "vxlan-tunnel"
```

Comment:
can we just call this one "submariner" as we do with wireguard?

Then it's just a standard in documentation and we don't need to handle special cases.
```suggestion
	VxlanIface             = "submariner"
```

Reply:
You also need to add an entry here:

https://github.com/submariner-io/submariner/blob/devel/pkg/routeagent_driver/handlers/ovn/handler.go#L80

For ovn to work, may be we can just standardize it to look for "submariner" interface regardless of the cable driver.

Reply:
Making it submariner cause issues when we support different cable drivers per link. Also when I tried changing it "submariner" , the vxlan-cleanup-driver in route agent , which does not check for the cable driver type , was deleting wireguard interface.

Reply:
> can we just call this one "submariner" as we do with wireguard?
> 
> 
> 
> Then it's just a standard in documentation and we don't need to handle special cases.
> 
> ```suggestion
> 
> 	VxlanIface             = "submariner"
> 
> ```

Making it submariner cause issues when we support different cable drivers per link. Also when I tried changing it "submariner" , the vxlan-cleanup-driver in route agent , which does not check for the cable driver type , was deleting wireguard interface.

Reply:
Ohh, ok, what you're saying makes sense.